httpconnection.go

```diff
@@ -41,27 +42,29 @@ func NewHTTPConnection(ctx context.Context, address string, options ...func(*htt
 	httpConn := &httpConnection{}
 
 	for _, option := range options {
-		option(httpConn)
+		if option != nil {
+			option(httpConn)
```

Comment:
can you please add an error check

Reply:
arrrrgh- missed this one. coming in new pr